app/helpers/application_helper.rb

```diff
@@ -22,4 +22,11 @@ def render_flash_message(flash_value)
     end
   end
 
+  def user_has_membership_application?
+    current_user.try(:membership_applications).try(:any?) && current_user.try(:membership_applications).try(:last).try(:persisted?)
+  end
+
+  def member_has_company?
+    current_user.is_member && !current_user.membership_applications.last.company.nil?
+  end
```

Comment:
OK, makes sense.

BTW, I see that it is optional for an application to have an associated company.  I thought that a user had to be associated with a company for an application to be approved (?).  (still learning the underlying business logic here ....)